src/cmt_decode_prometheus.c

```diff
@@ -200,7 +200,7 @@ static int parse_uint64(const char *in, uint64_t *out)
     int64_t val;
 
     errno = 0;
-    val = strtol(in, &end, 10);
+    val = strtoll(in, &end, 10);
```

Comment:
Should we use ifdef WIN32 here like as?

```c
#ifdef _WIN32
    val = strtoll(in, &end, 10);
#else
    val = strtol(in, &end, 10);
#endif 
```

I'm worrying about a side effect from moving `strtol` to `strtoll`.

Reply:
To confirm, I believe even current "64-bit" Windows has this long-is-32-bits setup. I don't know what the `_WIN32` definition comes from but you wouldn't want it to only apply to 32-bit Windows builds (but perhaps Win32 just means Windows?)

Reply:
Yes.

> _WIN32 Defined as 1 when the compilation target is 32-bit ARM, 64-bit ARM, x86, or x64. Otherwise, undefined.

ref: https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170

Reply:
@cosmo0920 probably a good idea to `_Static_assert` any assumptions relating to word sizes.
We found it generally helpful in our C codebases at my previous gig.

Reply:
There is some more Microsoft doco here on the topic of their C/C++ data type sizes, for reference: https://learn.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=msvc-170

Reply:
Also thanks for that tip @isubasinghe .

@cosmo0920 : `_Static_assert` is a C11 feature, see here: https://en.cppreference.com/w/c/language/_Static_assert . Does the Fluentbit codebase have any standards set regarding what C standard it targets?

Also, I can't see static assert being used anywhere in any Fluentbit code itself, however it does show up in several of the libraries Fluentbit uses: https://github.com/search?q=repo%3Afluent%2Ffluent-bit+static_assert&type=code

`mpack` is one of the big ones, with ~50 lines dedicated to properly `#define`ing it... https://github.com/ludocode/mpack/blob/79d3fcd3e04338b06e82d01a62f4aa98c7bad5f7/src/mpack/mpack-platform.h#L1418

For what it's worth, I bet `mpack` in particular is a library that absolutely has to get data type sizes exactly correct across platforms, so it might be one of the more comprehensive examples to draw a solution from.

Reply:
There's no need for a preprocessor check, I have verified the documentation and reality [using godbolt](https://godbolt.org/z/rbcG9T98M) and `long long` is a `QWORD` just as `int64_t` across architectures and compilers (32 bit armv7, 64 bit armv8, x86 and x86_64 using gcc, clang and msvc when possible).

So my take is that we can confidently use `strtoll` but feel free to verify my results.